### PR TITLE
Change product name per new guidelines

### PR DESCRIPTION
**Pull request by amudukutore**
Problem:
Product names were not meeting F5 guidelines.

Solution:
Changed product name references in all documentation per guidelines
using templates when available.

### CODE REVIEW

File: docs/README.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
-F5 Marathon BIG-IP Controller
-=============================
+|project|
```
Comment (jputrino) [inline docs/README.rst]: We shouldn't use substitutions for document titles because the substitution shows up in google searches.

**Comment by amudukutore** [comment]
Addressed code review comments

File: docs/RELEASE-NOTES.rst [hunk 2]
```diff
@@ -1,5 +1,5 @@
-Release Notes for Marathon BIG-IP Controller
-============================================
+Release Notes for |project|
```
Comment (jputrino) [inline docs/RELEASE-NOTES.rst]: This one shouldn't use the substitution in the doc title as well.

File: docs/conf.py [hunk 3]
```diff
@@ -83,7 +83,7 @@
 master_doc = 'index'
 
 # General information about the project.
-project = u'F5 Marathon BIG-IP Controller'
+project = u'F5 BIG-IP Controller for Marathon'
 copyright = u'2016,2017 F5 Networks Inc'
```
Comment (jputrino) [inline docs/conf.py]: Separate issue, but I think copyright can just be 2017. I'll double-check this.

Comment (amudukutore) [inline docs/conf.py]: I think we should list all years that the product has existed in the copyright. Also, it is a separate change from the rename effort.

File: docs/RELEASE-NOTES.rst (hunk 2 above)
Comment (amudukutore) [inline docs/RELEASE-NOTES.rst]: Good catch! Will fix.

**Comment by jputrino** [comment]
Fine to merge when build passes.

**Pull request merged**